File: doheeahn/231212/PGS_전화번호 목록.java
import java.util.*;
class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        
        HashSet<String> set = new HashSet<>();
        
        for(int i=0;i<phone_book.length;i++){
            set.add(phone_book[i]);
        }
        
        for(int i=0;i<phone_book.length;i++){
            for(int j=0;j<phone_book[i].length();j++){
                String head = phone_book[i].substring(0,j);
                if(set.contains(head)){
                    return false;
                }
            }
        }
        return answer;
    }
}

File: daehwa_park/2주차/PGS_전화번호_목록.py
def solution(phone_book):
    answer = True
    
    hash_map = {}
    
    for phone_num in phone_book:
        hash_map[phone_num] = 1
    for phone_num in phone_book:
        temp = ""
        for number in phone_num:
            temp += number
            if temp in hash_map and phone_num != temp:
                answer = False
                break
        if not answer:
            break
    
    return answer